Reflections/Week_2_reflection.md

```diff
@@ -0,0 +1,21 @@
+## Accomplishments
+
+The previous week has been rather slow. I spent most of my time working on building my program on windows for windows users to test my current project, however I was blocked significantly by linking issues. I also started a branch reworking the beizer curve implementation in order to reduce possible future errors.
+
+## Challenges
+
+The biggest challenge was that attempting to build the program on windows. I had attempted to build the program in a windows VM, using both MSVC and QtCreator, with many different modifications to my CMake files. Nothing that I attempted worked. I would either be unable to link correctly to the Qt libraries, or unable to properly link to the FastNose2 libraries. After about 4 concerted days of work on this, I had to give up, as I could not get it to work at all.
+
+Additionally, I had hoped to figure out how to efficently draw equidistant subdivisions of beizer curves, however so far the math has been impossible for me to properly understand.
```

Comment:
What is meant by "equidistant subdivisions of Bezier curves"? We might be able to tackle that one together, depending on what the goal is here.

Reply:
To try and say it as clearly as possible:

A beizer curve is a parametric function f(t) over the range [0,1].

the distance moved by the point f(t) from t=x to t=x+n is not constant for all values x for a given n. 

I would like to have a mapping of inputs g such that f(g(t)) - f(g(t+n)) is proportional to n. That way the spacing of points along the curve does not become closer or more distant over the length of the line.  

![image](https://user-images.githubusercontent.com/12928246/115155936-111eef80-a050-11eb-92ac-fa0ff118697c.png)
An example of the inconsistent spacing issue that I would want to resolve.

Reply:
With the explanation you've given, I was able to research relevant materials and put together an example. I've used some parts of your code, only translated for use with CTurtle, so it should seem relatively familiar-ish. This operation is called "flattening", as it "flattens" the line depending on the requested number of samples. Fewer samples from the bezier curve, flatter set of line segments.

25 bezier samples to 5 flattened points:
![Flattening Example](https://i.imgur.com/SXZRAv4.png)

25 bezier samples to 10 flattened points:
![Flattening Exmaple 2](https://i.imgur.com/9Pyigkp.png)

[The Gist for this example can be found here](https://gist.github.com/walkerje/422ec7e7470791a788d7a2bf41dc5397)
It is based on a StackOverflow solution which I found relatively difficult to read (formal language is not my forte), however it is also linked in the Gist solution.